tests/dbt/test_graph.py

```diff
@@ -429,7 +432,10 @@ def test_load_via_dbt_ls_does_not_create_target_logs_in_original_folder(
 
     used_cwd = Path(mock_popen.call_args[0][0][-5])
     assert used_cwd != project_config.dbt_project_path
-    assert not used_cwd.exists()
+    # When the cache profile is enabled, the profile is created in {cache_dir}/profile/{version}/profiles.yml
+    # rather than in a temporary file. This ensures it is persisted for future use and remains uncleared.
+    if not enable_cache_profile:
+        assert not used_cwd.exists()
```

Comment:
@tatiana, `used_cwd` pointing to the dir where profiles.yml exists. When cache profile is enabled, it is created inside `cache_dir/profile/version`, ensuring persistence. If profile cache is enabled, does not it make sense, or I'm still missing something?

CI log:
```
FAILED tests/dbt/test_graph.py::test_load_via_dbt_ls_does_not_create_target_logs_in_original_folder[True] - AssertionError: assert not True
 +  where True = <bound method Path.exists of PosixPath('/tmp/cosmos/profile/470f1452ea88562203ec89a49bab7e2c6043ef3b7f93e61ce30772e24aa38289')>()
 +    where <bound method Path.exists of PosixPath('/tmp/cosmos/profile/470f1452ea88562203ec89a49bab7e2c6043ef3b7f93e61ce30772e24aa38289')> = PosixPath('/tmp/cosmos/profile/470f1452ea88562203ec89a49bab7e2c6043ef3b7f93e61ce30772e24aa38289').exists
```
Ref CI: https://github.com/astronomer/astronomer-cosmos/actions/runs/9679063910/job/26704228626